index.js

```diff
@@ -0,0 +1,124 @@
+const URL = 'https://trektravel.herokuapp.com/trips';
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportApiError = (error) => {
+  let errors = Object.values(error)[2].data.errors;
+  let errorHtml = `<p>${error.message}</p><ul>`;
+
+  for (let [field, problem] of Object.entries(errors)) {
+    errorHtml += `<li>${field}: ${problem}</li>`;
+  }
+  errorHtml += '</ul>';
+  reportStatus(errorHtml);
+}
+
+const loadTripForm = (tripName) => {
+  const tripForm = $('#trip-form');
+  tripForm.addClass('card');
+  tripForm.append(
+    `<h4 class="card-header">Reserve a spot on ${tripName}</h4>
+    <div class="card-body">
+      <div class="form-group">
+        <label for="name">Name</label>
+        <input type="text" name="name" class="form-control" placeholder="Your Name">
+      </div>
+      <div class="form-group">
+        <label for="email">Email</label>
+        <input type="text" name="email" class="form-control" placeholder="you@example.com">
+      </div>
+      <div>
+        <input type="submit" class="btn btn-primary" name="reserve-trip" value="Reserve">
+      </div>
+    </div>`
+  )
+}
+
+const loadTripDetail = (response) => {
+  const tripDetail = $('#trip-details');
+  tripDetail.empty();
+  tripDetail.addClass('card');
+  tripDetail.append(
+    `<h4 class="card-header">Trip Details</h4>
+    <div class="card-body">
+      <h5 class="card-title">Name: ${response.data.name}</h5>
+      <p><strong>Continent:</strong> ${response.data.continent}</p>
+      <p><strong>Category:</strong> ${response.data.category}</p>
+      <p><strong>Weeks:</strong> ${response.data.weeks}</p>
+      <p><strong>Cost:</strong> $${response.data.cost}</p>
+      <p><strong>About:</strong></p>
+      <p>${response.data.about}</p>
+    </div>`
+  );
+}
+
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+
+  axios.get(URL)
+    .then((response) => {
+      tripList.addClass('card');
+      tripList.append(`<h4 class="card-header">All Trips</h4>`);
+      $('h4').addClass('trip-section-title');
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+
+      response.data.forEach((trip) => {
+        tripList.append(`<button id=${trip.id}>${trip.name}</button>`);
+        $(`#${trip.id}`).addClass('btn btn-light btn-block');
+        $(`#${trip.id}`).click( function() {
+          $('#status-message').empty();
+          tripDetails($(this).attr('id'));
+          $('#trip-form').empty();
+          loadTripForm(trip.name);
+          $('#trip-form').submit((event) => {
```

Comment:
Every time I click on a trip to see details , this function is adding an event listener to the form.  **EVERY** a trip is clicked another event handler is added.  The old ones don't go away.  So if I click around and then submit the form the app will submit **multiple** `POST` requests to the API.  So I'll find myself registered multiple times for the same trip.

Reply:
Thanks, Chris! I fixed this bug by adding the .off() method to the form. So all the event handlers attached previously were removed.